### PR TITLE
Use a common FNET_Transport owned by Proton in both SceduledExecutor …

### PR DESCRIPTION
**Pull request by baldersheim**
…and TransactionLogServer.

This reduces the number of Transport object by 1 per document type and netto 1 in Proton.
Each of them contains 2 threads.

In addition it uses a common Transport for the RpcFileAcquirer objects used during config fetching.
This prevents creating 3 temporary Transport objects on every reconfig.

@geirst @geirst PR
@havard @vekterli FYI

### CODE REVIEW

**Review by geirst — APPROVED** [review]
👍

**Review by geirst — APPROVED** [review]
👍

File: searchcore/src/apps/vespa-transactionlog-inspect/vespa-transactionlog-inspect.cpp [hunk 1]
```diff
@@ -371,17 +373,24 @@ class BaseUtility : public Utility
...
           _fileHeader(),
-          _server(_bopts.tlsName, _bopts.listenPort, _bopts.tlsDir, _fileHeader),
+          _threadPool(64_Ki),
+          _transport(),
+          _server(_transport, _bopts.tlsName, _bopts.listenPort, _bopts.tlsDir, _fileHeader),
           _client(vespalib::make_string("tcp/localhost:%d", _bopts.listenPort))
     {
     }
+    ~BaseUtility() override {
+        _transport.ShutDown(true);
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply searchcore/src/apps/vespa-transactionlog-inspect/vespa-transactionlog-inspect.cpp:393]: It was not, good catch, fixed.

File: searchcore/src/tests/proton/documentdb/fileconfigmanager/fileconfigmanager_test.cpp [hunk 2]
```diff
@@ -125,55 +128,69 @@ addConfigsThatAreNotSavedToDisk(const DocumentDBConfig &cfg)
     return builder.build();
 }
 
-TEST_F("requireThatConfigCanBeSavedAndLoaded", DocumentDBConfig::SP(makeBaseConfigSnapshot()))
+struct Fixture {
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply searchcore/src/tests/proton/documentdb/fileconfigmanager/fileconfigmanager_test.cpp]: Replaced with TransportMgr helper class.

File: config/src/tests/file_acquirer/file_acquirer_test.cpp [hunk 3]
```diff
@@ -3,12 +3,16 @@
...
 #include <vespa/fnet/frt/rpcrequest.h>
+#include <vespa/fnet/transport.h>
+#include <vespa/fastos/thread.h>
 #include <vespa/vespalib/util/stringfmt.h>
 
 using namespace config;
 
 struct ServerFixture : FRT_Invokable {
     fnet::frt::StandaloneFRT server;
+    FastOS_ThreadPool threadPool;
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply config/src/tests/file_acquirer/file_acquirer_test.cpp:14]: I have added a helper class, TransportMgr in searchcore/proton/test/transport_helpers.h.
Not sure if we should keep it there or move it to a more common place.

[the remark this answers is not in the document]
Comment (baldersheim) [reply config/src/tests/file_acquirer/file_acquirer_test.cpp:14]: Keeping it here for now,

**Pull request merged**